modules/file_upload/cms_file_uploader_module.e

```diff
@@ -0,0 +1,338 @@
+note
+	description: "file_upload application root class"
+	date: "$Date$"
+	revision: "$Revision$"
+
+class
+	CMS_FILE_UPLOADER_MODULE
+
+inherit
+	CMS_MODULE
+		rename
+			module_api as file_upload_api
+		redefine
+			install,
+			initialize,
+			setup_hooks,
+			permissions,
+			file_upload_api
+		end
+
+	CMS_HOOK_MENU_SYSTEM_ALTER
+
+	SHARED_EXECUTION_ENVIRONMENT
+
+create
+	make
+
+feature {NONE} -- Initialization
+
+	make
+		do
+			name := "file_uploader"
+			version := "1.0"
+			description := "Service to upload files, and manage them."
+			package := "file"
+		end
+
+feature -- Access
+
+	name: STRING
+
+	permissions: LIST [READABLE_STRING_8]
+			-- List of permission ids, used by this module, and declared.
+		do
+			Result := Precursor
+			Result.force ("admin uploaded files")
+			Result.force ("upload files")
+		end
+
+feature {CMS_API} -- Module Initialization
+
+	initialize (api: CMS_API)
+			-- <Precursor>
+		do
+			Precursor (api)
+			if file_upload_api = Void then
+				create file_upload_api.make (api)
+			end
+		end
+
+feature {CMS_API}-- Module management
+
+	install (api: CMS_API)
+			-- install the module
+		local
+			sql: STRING
+			l_file_upload_api: like file_upload_api
+			d: DIRECTORY
+		do
+				-- create a database table
+			if attached {CMS_STORAGE_SQL_I} api.storage as l_sql_storage then
```

Comment:
for now, you don't use any database, so either you remove those lines, or you prepare the interface that will access the db.